packages/bff/src/services/catalogService.ts

```diff
@@ -119,8 +198,86 @@ export function catalogServiceBuilder(
           totalCount: eservicesResponse.totalCount,
         },
       };
-
       return response;
     },
+    getProducerEServiceDescriptor: async (
+      eserviceId: EServiceId,
+      descriptorId: DescriptorId,
+      context: WithLogger<BffAppContext>
+    ): Promise<bffApi.ProducerEServiceDescriptor> => {
+      const requesterId = context.authData.organizationId;
+      const headers = context.headers;
+
+      const eservice: catalogApi.EService =
+        await catalogProcessClient.getEServiceById({
+          params: {
+            eServiceId: eserviceId,
+          },
+          headers,
+        });
+
+      if (eservice.producerId !== requesterId) {
+        throw invalidEServiceRequester(eserviceId, requesterId);
+      }
+
+      const descriptor = getEserviceDesciptor(eservice, descriptorId);
+
+      const descriptorAttributeIds = getAttributeIds(descriptor);
+
+      const attributes = await getBulkAttributes(
+        attributeProcessClient,
+        headers,
+        descriptorAttributeIds
+      );
+
+      const descriptorAttributes = {
+        certified: [
+          toBffCatalogApiDescriptorAttribute(
+            attributes,
+            descriptor.attributes.certified.flat()
+          ),
+        ],
+        declared: [
+          toBffCatalogApiDescriptorAttribute(
+            attributes,
+            descriptor.attributes.declared.flat()
+          ),
+        ],
+        verified: [
+          toBffCatalogApiDescriptorAttribute(
+            attributes,
+            descriptor.attributes.verified.flat()
+          ),
+        ],
+      };
```

Comment:
This could also become a function `toBffCatalogApiDescriptorAttributes(descriptorAttributes)`